### PR TITLE
Parser fixes

### PR DESCRIPTION
**Pull request by mkurc-ant**
- Fixed parsing of "real triple" where there can be 1, 2, 3 or zero numbers.
  - According to the SDF specification a delay value is expressed using "read triples". These express min, avg and max delay. However the standard allows not to specify all thee of therm, some fields may be missing. That was not supported by the parser.
- Fixed parsing of delay specification which may now consist of 1, 2 or 3 "real triples"
  - Depending on the complexity of the delay model, there may be multiple "real triples" per delay entry. With this PR multiple cases (unfortunately not all of them) are supported.
- The SDF writer no longer enforces upper case for CELLNAME
- Added tests.

### CODE REVIEW

File: sdf_timing/sdfyacc.py [hunk 1]
```diff
@@ -380,21 +380,35 @@ def p_del(p):
...
+        paths['fast'] = p[1]
+        paths['slow'] = p[2]
+    elif len(p) == 4:
+        paths['fast'] = p[1]
+        paths['nominal'] = p[2]
+        paths['slow'] = p[3]
+    p[0] = paths
+
+def p_device(p):
+    'device : LPAR DEVICE port_spec delval_list RPAR'
```
Comment (mithro) [inline sdf_timing/sdfyacc.py]: These should all be triple quoted strings, not single quotes.

File: sdf_timing/sdfyacc.py [hunk 2]
```diff
@@ -380,21 +380,35 @@ def p_del(p):
...
 
-def p_device(p):
-    'device : LPAR DEVICE port_spec real_triple real_triple RPAR'
+def p_delval_list(p):
+    '''delval_list : real_triple
+                   | real_triple real_triple
+                   | real_triple real_triple real_triple'''
+
+    # TODO: There should be more possibilities up to 12 real_triples
+    # But we stick with 1, 2 or 3 for now.
```
Comment (mithro) [inline sdf_timing/sdfyacc.py]: Why are you sticking to 1/2/3 for now?

File: sdf_timing/sdfwrite.py [hunk 3]
```diff
@@ -233,7 +233,7 @@ def emit_sdf(timings, timescale='1ps'):
                 for location in sorted(timings['cells'][cell]):
                     sdf += """
     (CELL
-        (CELLTYPE \"{name}\")""".format(name=cell.upper())
+        (CELLTYPE \"{name}\")""".format(name=cell)
```
Comment (mithro) [inline sdf_timing/sdfwrite.py]: Why are you removing the `upper()`?

Comment (mkurc-ant) [reply sdf_timing/sdfwrite.py]: According to the SDF specification the CELLTYPE does not have to be uppercase. And before that change the round trip SDF -> python_sdf_parse -> SDF caused all these to be made uppercase.

File: sdf_timing/sdfyacc.py (hunk 2 above)
Comment (mkurc-ant) [reply sdf_timing/sdfyacc.py]: The SDF can express separate timings from transitions between different logic state sets. For now we do not have use for eg. different 0->1 and 1->0 delays. Thats' why I limited parsing to only that triple count that we need.

File: sdf_timing/sdfyacc.py (hunk 1 above)
Comment (mkurc-ant) [reply sdf_timing/sdfyacc.py]: Fixed.

File: sdf_timing/sdfyacc.py (hunk 2 above)
Comment (mithro) [reply sdf_timing/sdfyacc.py]: Please add the following info to the comment.

File: sdf_timing/sdfwrite.py (hunk 3 above)
Comment (mithro) [reply sdf_timing/sdfwrite.py]: It would probably good to have a "uppercase everything" mode?

File: sdf_timing/sdfyacc.py (hunk 2 above)
Comment (mkurc-ant) [reply sdf_timing/sdfyacc.py]: Added.

File: sdf_timing/sdfwrite.py (hunk 3 above)
Comment (mkurc-ant) [reply sdf_timing/sdfwrite.py]: That makes sense. I've added an additional parameter to the `emit_sdf` function.

**Pull request merged**